Tests/ArcGISToolkitTests/AuthenticationManagerTests.swift

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+import ArcGIS
+@testable import ArcGISToolkit
+
+@MainActor final class AuthenticationManagerTests: XCTestCase {
+    func testHandleChallengesUsingAuthenticator() {
+        addTeardownBlock {
+            ArcGISEnvironment.authenticationManager.arcGISAuthenticationChallengeHandler = nil
+            ArcGISEnvironment.authenticationManager.networkAuthenticationChallengeHandler = nil
+        }
+        
+        let authenticator = Authenticator()
+        ArcGISEnvironment.authenticationManager.handleChallenges(using: authenticator)
+        XCTAssertNotNil(ArcGISEnvironment.authenticationManager.arcGISAuthenticationChallengeHandler)
+        XCTAssertNotNil(ArcGISEnvironment.authenticationManager.networkAuthenticationChallengeHandler)
```

Comment:
you can assert that it's equal to the authenticator, right?

Reply:
The handles are not `Equatable`. Shall we make them?

Reply:
no but you can cast to an authenticator I think with `as?` right?

Reply:
yes, I can do `XCTAssertIdentical` using `as?`.

```
        XCTAssertIdentical(ArcGISEnvironment.authenticationManager.arcGISAuthenticationChallengeHandler as? Authenticator, authenticator)
        XCTAssertIdentical(ArcGISEnvironment.authenticationManager.networkAuthenticationChallengeHandler as? Authenticator, authenticator)
```